ptf/saiswitch.py

```diff
@@ -125,64 +104,87 @@ def availableIPv4RouteEntryTest(self):
+        self.nhop = sai_thrift_create_next_hop(
+            self.client,
+            ip=sai_ipaddress('10.10.10.1'),
+            router_interface_id=self.port10_rif,
+            type=SAI_NEXT_HOP_TYPE_IP)
+        self.assertNotEqual(self.nhop, SAI_NULL_OBJECT_ID)
+
+        self.route_number = 0
+        max_host_route = 0
+        self.switch_helper.create_route_entry_from_default_vrf(self)
+        while self.route_number < max_route_entry:
+            ip_p_m = sai_ipprefix(next(ip_add) + mask)
+
+            # check if ip repeat, then get next ip
+            if str(ip_p_m) in self.routes:
+                continue
+
+            route_entry = sai_thrift_route_entry_t(
+                vr_id=self.default_vrf,
+                destination=ip_p_m)
+            status = sai_thrift_create_route_entry(
+                self.client, route_entry, next_hop_id=self.nhop)
+
+            if status == SAI_STATUS_SUCCESS:
+                self.routes.update({str(ip_p_m): route_entry})
+                self.route_number += 1
+            elif status == SAI_STATUS_ITEM_ALREADY_EXISTS:
+                continue
+            elif mask == '/32':  # when host table is full change to LPM
+                print("%s host routes have been created" % self.route_number)
+                max_host_route = self.route_number
+                self.available_v4_host_routes = max_host_route
+                mask = '/30'
+                continue
+            else:
+                self.fail("Route creation failed after creating %d "
+                            "entries, status %u" % (self.route_number, status))
+
+            attr = sai_thrift_get_switch_attribute(
+                self.client, available_ipv4_route_entry=True)
+            self.assertEqual(attr["available_ipv4_route_entry"],
+                                max_route_entry - self.route_number)
+        ip_add.close()
 
-                attr = sai_thrift_get_switch_attribute(
-                    self.client, available_ipv4_route_entry=True)
-                self.assertEqual(attr["available_ipv4_route_entry"],
-                                 max_route_entry - route_number)
 
-            print("%s LPM routes have been created"
-                  % (max_route_entry - max_host_route))
-            self.assertEqual(attr["available_ipv4_route_entry"], 0)
+    def tearDown(self):
+        for ip_p_m in self.routes:
+            sai_thrift_remove_route_entry(self.client, self.routes.get(ip_p_m))
+        sai_thrift_remove_next_hop(self.client, self.nhop)
 
-            ip_add.close()
 
-        finally:
-            for ip_p_m in routes:
-                sai_thrift_remove_route_entry(self.client, routes.get(ip_p_m))
-            sai_thrift_remove_next_hop(self.client, nhop)
+    def check_amount(self):
+        attr = sai_thrift_get_switch_attribute(
+            self.client, available_ipv4_route_entry=True)
+        self.assertEqual(attr["available_ipv4_route_entry"], 0)
+
+
+    def runTest(self):
```

Comment:
minor: In all our test files we tried to keep test classes methods in the same order: setUp, runTest, tearDown, and then the rest of test methods.